lib/discourse_topic_voting/categories_controller_extension.rb

```diff
@@ -4,7 +4,7 @@ module DiscourseTopicVoting
   module CategoriesControllerExtension
     def category_params
       @vote_enabled ||=
-        !!ActiveRecord::Type::Boolean.new.cast(params[:custom_fields]&.delete(:enable_topic_voting))
```

Comment:
this `delete()` here removes the parameter from the core controller, I guess it's because we don't want to store the custom field (because we have a separate table for the category/setting mappings) but it also means that it skips core's default handling for the staff action logger. 

Keeping the `enable_topic_voting` param keeps the staff action logging from core, but it also stores a row in the category custom fields table for `enable_topic_voting: true`.

Reply:
I'm not too familiar with this part of the code and I don't know the history of moving from the custom field, but I feel like if our best practice is having a separate table, we could have a better DX to be able to pass the value from the FE and logging the staff action without relying on a custom field. 

But it's a bigger conversation, I don't think having the custom field row will cause any issue.